tests/sqlparser_common.rs

```diff
@@ -11460,3 +11460,54 @@ fn test_try_convert() {
         all_dialects_where(|d| d.supports_try_convert() && !d.convert_type_before_value());
     dialects.verified_expr("TRY_CONVERT('foo', VARCHAR(MAX))");
 }
+
+#[test]
+fn parse_bang_not() {
```

Comment:
since this MR also introduces the `supports_factorial_operator` dialect method, can we add a similar test case for that one as well?